src/main/java/fi/aalto/cs/apluscourses/presentation/exercise/SubmissionViewModel.java

```diff
@@ -104,6 +108,25 @@ public String getSubmissionCountText() {
     return submissionCountText.toString();
   }
 
+  /**
+   * Formats a descriptive string for the submission dialog about a submittable file. The string
+   * includes the file name and its last modification date.
+   *
+   * @param file The submittable file in question.
+   */
+  @NotNull
+  public String getFileInformationText(SubmittableFile file) {
+    StringBuilder fileInfoText = new StringBuilder(file.getName());
+    try {
+      String lastModificationTime =
+              FileDateFormatter.getFileModificationTime(filePaths.get(file.getKey()));
+      fileInfoText.append(" (modified ").append(lastModificationTime).append(")");
+    } catch (IOException e) {
+      // don't print anything about the last modification time if a very unlikely exception happened
```

Comment:
```java
//top level declaration
private static final Logger logger = LoggerFactory.getLogger(AboutAction.class);

//instead of ioException.printStackTrace();
logger.error("An exception was thrown for some mysterious reason", e);
```